### PR TITLE
Restructure code

### PR DESCRIPTION
**Pull request by deyarchit**
## Summary by CodeRabbit

* **Refactor**
  * Disk operations were reorganized into a class-based interface, improving structure and safety for file and directory management.
  * Internal code updated to use the new class-based approach for disk operations.
* **Chores**
  * Updated test suite to use the new class interface and improved test coverage for edge cases.
  * Modified Makefile with new targets for integration data generation and pre-release checks.
  * Adjusted `.gitignore` to exclude integration test data from version control.
* **New Features**
  * Added a utility for generating short SHA-256 hashes for files.

### CODE REVIEW

File: organizer/disk_operations.py [hunk 1]
```diff
@@ -1,183 +1,163 @@
...
+        self._delete_empty_dirs(missing)
+
+    def _create_directories(self, added: List[FlatFileItem]) -> None:
+        for item in added:
+            if item.path.endswith("/"):
+                full_path = self._to_abs(item.path)
+                if not os.path.exists(full_path):
+                    print(f"Creating directory: {full_path}")
+                    os.makedirs(full_path, exist_ok=True)
+                self.handled_paths.add(item.path)
```
[the remark this answers is not in the document]
Comment (deyarchit) [reply organizer/disk_operations.py:33]: @coderabbitai resolve
